src/buffer.h

```diff
@@ -45,17 +45,37 @@ struct mpd_buffer {
 	unsigned read;
 
 	/** the actual buffer */
-	unsigned char data[4096];
+	unsigned char *data;
+
+	/** size of buffer */
+	size_t data_len;
 };
 
 /**
  * Initialize an empty buffer.
  */
-static inline void
+static inline void *
 mpd_buffer_init(struct mpd_buffer *buffer)
 {
+	const size_t std_data_len = 4096;
+
 	buffer->read = 0;
 	buffer->write = 0;
+	buffer->data_len = std_data_len;
+	buffer->data = malloc(buffer->data_len);
+	if (buffer->data != NULL)
+		memset(buffer->data, 0, buffer->data_len);
```

Comment:
Why `memset()`? It wasn't here before. Why is this suddenly necesarry?

Reply:
It is not necessary; it is just a precaution to initialize the data to a known state